code/modules/ninja/suit/ninjaDrainAct.dm

```diff
@@ -44,7 +44,7 @@ They *could* go in their appropriate files, but this is supposed to be modular
 			if (do_after(H,10, target = src))
 				spark_system.start()
 				playsound(loc, "sparks", 50, 1)
-				cell.charge -= drain
+				cell.use(drain)
 				S.cell.charge += drain
```

Comment:
S.cell.give(drain)

Reply:
cells run on the give proc? want me to mirror this for all of the drains as almost all of the drain procs give his cell charge?

Reply:
there's 8 more places that need to be updated too, counting this one.

Reply:
Yeah, and hit the rest of the `cell.charge -= drian`'s in this file too please.

Reply:
give is for gain in charge (S.cell is the ninja's battery) and cell is the source item's cell.

You should use use/give instead of basic += -= to avoid similar situations with other items.

Reply:
It isn't thatttttt big an issue since we've done the math above to keep from draining and adding more than their cell can take, but use() and give() for cells have additional logic related to exploding if they have a bomb attached to them that's wired up to explode on energy changes.

Reply:
drian? but i barely knew her!

alright i'm on it

Reply:
don't say that lol

Reply:
the knew her joke or the maintaining code one